Pylint: Fix unnecessary-dunder-call issues.

## 💡 Description
Update the codebase to not use calls to `__len__` or `__next__` magic methods. pylint report is attached (`pylint . --disable=all --enable=unnecessary-dunder-call`):
```
************* Module pds.naif_pds4_bundler.classes.collection.collection_kernels
pds/naif_pds4_bundler/classes/collection/collection_kernels.py:124:20: C2801: Unnecessarily calls dunder method __len__. Use len built-in function. (unnecessary-dunder-call)
pds/naif_pds4_bundler/classes/collection/collection_kernels.py:125:24: C2801: Unnecessarily calls dunder method __len__. Use len built-in function. (unnecessary-dunder-call)
pds/naif_pds4_bundler/classes/collection/collection_kernels.py:126:24: C2801: Unnecessarily calls dunder method __len__. Use len built-in function. (unnecessary-dunder-call)
pds/naif_pds4_bundler/classes/collection/collection_kernels.py:127:24: C2801: Unnecessarily calls dunder method __len__. Use len built-in function. (unnecessary-dunder-call)
************* Module pds.naif_pds4_bundler.classes.product.product_metakernel
pds/naif_pds4_bundler/classes/product/product_metakernel.py:851:25: C2801: Unnecessarily calls dunder method __len__. Use len built-in function. (unnecessary-dunder-call)
************* Module pds.naif_pds4_bundler.classes.product.product_orbnum
pds/naif_pds4_bundler/classes/product/product_orbnum.py:844:34: C2801: Unnecessarily calls dunder method __next__. Use next built-in function. (unnecessary-dunder-call)

```

--- a/src/pds/naif_pds4_bundler/classes/collection/collection_kernels.py
+++ b/src/pds/naif_pds4_bundler/classes/collection/collection_kernels.py
@@ -121,10 +121,10 @@ class SpiceKernelsCollection(Collection):
                 and (self.setup.args.faucet != "labels")
             ):
                 if (
-                    self.setup.mk.__len__() == 1
-                    and self.setup.mk[0]["name"].__len__() == 1
-                    and self.setup.mk[0]["name"][0].__len__() == 1
-                    and self.setup.mk[0]["name"][0]["pattern"].__len__() == 2
+                    len(self.setup.mk) == 1
+                    and len(self.setup.mk[0]["name"]) == 1
+                    and len(self.setup.mk[0]["name"][0]) == 1
+                    and len(self.setup.mk[0]["name"][0]["pattern"]) == 2
                     and not isinstance(self.setup.mk[0]["name"][0]["pattern"], list)
                 ):
                     if self.setup.mk[0]["name"][0]["pattern"]["#text"] == "VERSION":

--- a/src/pds/naif_pds4_bundler/classes/product/product_metakernel.py
+++ b/src/pds/naif_pds4_bundler/classes/product/product_metakernel.py
@@ -848,7 +848,7 @@ class MetaKernelProduct(Product):
             # we need to subtract 1 kernel.
             #
             ker_num_fr = spiceypy.ktotal("ALL") - 1
-            ker_num_mk = self.collection_metakernel.__len__()
+            ker_num_mk = len(self.collection_metakernel)
 
             logging.info(f"-- Kernels loaded with FURNSH: {ker_num_fr}")
             logging.info(f"-- Kernels present in {self.name}: {ker_num_mk}")

--- a/src/pds/naif_pds4_bundler/classes/product/product_orbnum.py
+++ b/src/pds/naif_pds4_bundler/classes/product/product_orbnum.py
@@ -841,7 +841,7 @@ class OrbnumFileProduct(Product):
             # To find blank spaces we use a regular expression iterator.
             #
             if number > 0:
-                blankspaces_loc = blankspace_iter.__next__().regs[0]
+                blankspaces_loc = next(blankspace_iter).regs[0]
                 blankspaces = blankspaces_loc[1] - blankspaces_loc[0] + 1
                 location += int(length) + blankspaces
 

